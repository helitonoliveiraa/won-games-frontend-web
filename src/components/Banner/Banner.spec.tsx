import { screen } from '@testing-library/react';

import { renderWithTheme } from '@/utils/tests/helpers';

import { Banner } from './index';

const bannerProps = {
  image: 'fake-image-url',
  title: 'fake-title',
  subtitle: 'fake-subtitle',
  buttonLabel: 'fake-buttonLabel',
  buttonLink: 'fake-buttonLink',
};

describe('<Banner/>', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Banner {...bannerProps} />);

    expect(
      screen.getByRole('heading', { name: /fake-title/i }),
    ).toBeInTheDocument();
    expect(screen.getByText(/fake-subtitle/i)).toBeInTheDocument();
    expect(
      screen.getByRole('img', { name: /fake-title/i }),
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
