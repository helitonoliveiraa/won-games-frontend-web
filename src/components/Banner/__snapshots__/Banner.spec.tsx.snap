// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner/> should render correctly 1`] = `
.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.4rem;
  background: linear-gradient(180deg,#FF5F5F 0%,#f062c0) 50%;
  color: #FAFAFA;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: background 0.3s ease-in-out;
  transition: background 0.3s ease-in-out;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c3:hover {
  background: linear-gradient(180deg,#e35563 0%,#d958a6);
}

.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #EAEAEA;
  background-image: url(fake-image-url);
  background-position: center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 1.6rem 2.4rem;
}

.c2 h2 {
  font-size: 2rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c2 h3 {
  font-size: 1.4rem;
  font-weight: 400;
  color: #FAFAFA;
  margin-bottom: 1.6rem;
}

.c2 h3 strong {
  font-weight: 700;
  color: #F231A5;
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.4rem rgba(0,0,0,0.25);
  }
}

@media (min-width:768px) {
  .c1 {
    height: 58.4rem;
  }
}

@media (min-width:768px) {
  .c2 {
    padding: 3.2rem 4.0rem;
    border-radius: 0 0 0.4rem 0.4rem;
    position: absolute;
    left: 0;
    right: 0;
    bottom: 0;
  }

  .c2 h2 {
    font-size: 2.8rem;
  }

  .c2 h3 {
    font-size: 2rem;
  }
}

<section
  class="c0"
>
  <div
    aria-label="fake-title"
    class="c1"
    role="img"
    src="fake-image-url"
  />
  <div
    class="c2"
  >
    <h2>
      fake-title
    </h2>
    <h3>
      fake-subtitle
    </h3>
    <a
      class="c3"
      href="fake-buttonLink"
    >
      <span>
        fake-buttonLabel
      </span>
    </a>
  </div>
</section>
`;
